lms/djangoapps/student_account/views.py

```diff
@@ -124,6 +124,22 @@ def login_and_registration_form(request, initial_mode="login"):
         } for message in messages.get_messages(request) if 'account-activation' in message.tags
     ]
 
+    account_creation_allowed = configuration_helpers.get_value(
+        'ALLOW_PUBLIC_ACCOUNT_CREATION', settings.FEATURES.get('ALLOW_PUBLIC_ACCOUNT_CREATION', True)
+    )
+
+    # add user details from running pipeline
+    pipeline_user_details = {}
+    running_pipeline = pipeline.get(request)
+    if running_pipeline:
+        pipeline_user_details = running_pipeline['kwargs']['details']
+        if 'email' in pipeline_user_details or 'username' in pipeline_user_details:
+            # pylint: disable=invalid-name
+            qs = {'email': pipeline_user_details['email']} if pipeline_user_details.get('email') else \
+                {'username': pipeline_user_details.get('username')}
+            if User.objects.filter(**qs).exists():
+                account_creation_allowed = False
```

Comment:
Move this code to `_third_party_auth_context`, populate the `third_party_auth` dict that is added to the context with the values that you need and pull them from there in the front-end code.

Reply:
If we do that we will have duplicate functionality to handle the same user case that `account_creation_allowed` handles. This way we are just overriding `account_creation_allowed` that makes sure of the correct behavior and we do not need to make any overrides in the front end code.